fido/prepare.py

```diff
@@ -351,7 +351,7 @@ def doByte(chars, i, littleendian, esc=True):
     c1 = '0123456789ABCDEF'.find(chars[i].upper())
     c2 = '0123456789ABCDEF'.find(chars[i + 1].upper())
     if (c1 < 0 or c2 < 0):
-        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars))
+        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars, buf=None))
```

Comment:
Is this tested? I have a horrible feeling that the `buf.getValue()` in `_convert_err_msg` will crash when `buf=None`. Something (very) like `AttributeError: 'NoneType' object has no attribute 'getvalue'`

Reply:
You're right -- sorry I should have put a [WIP] flag on this!!

Reply:
No problem